indra_gpt/pipelines/statement_extraction_pipeline.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import pickle
+from pathlib import Path
+
+class StatementExtractionPipeline:
+    def __init__(self, pre_processor, generator, post_processor):
```

Comment:
Here it can be helpful to add type annotations to make the code more clear:
```suggestion
    def __init__(self, pre_processor: PreProcessor, generator: Generator, post_processor: PostProcessor):
```